romanisim/tests/test_bandpass.py

```diff
@@ -64,10 +72,161 @@ def test_compute_abflux(filter):
     gauss_flux = bandpass.compute_abflux(data_table)
 
     # Comparing both fluxes as magnitudes
-    assert np.isclose(np.log10(theo_flux.value), np.log10(gauss_flux[filter]), atol=1.0e-6)
+    assert np.isclose(np.log10(theo_flux.value), np.log10(gauss_flux[filter]), rtol=1.0e-6)
 
 
 @pytest.mark.parametrize("filter, value", zip(FILTERLIST, ABVLIST))
 def test_get_abflux(filter, value):
     # Test that proper results (within 10%) are returned for select bands.
     assert np.isclose(bandpass.get_abflux(filter), value, rtol=1.0e-1)
+
+
+@metrics_logger("DMS233")
+@pytest.mark.soctests
+def test_convert_flux_to_counts():
+    # Define AB zero flux, and dirac delta wavelength
+    dd_wavel = 1.290 * u.micron
+    effarea = bandpass.read_gsfc_effarea()
+
+    # Create dirac-delta-like distribution for filter response
+    dd_wavedist = effarea['Wave'] * u.micron
+    wave_bin_width = dd_wavedist[1] - dd_wavedist[0]
+    thru = norm.pdf(dd_wavedist, dd_wavel.value, 0.001)
+
+    # Add constant flux
+    thru += 100
+
+    # Rescale
+    thru *= 1.0e-35
+
+    # Add flux units
+    thru *= u.erg / (u.s * u.cm ** 2 * u.hertz)
+
+    theo_flux = {}
+    theo_flux_sum = {}
+    gauss_flux = {}
+
+    for filter in IFILTLIST:
+        # Define filter area
+        area = bandpass.read_gsfc_effarea()[filter] * u.m ** 2
+
+        # Analytical flux
+        theo_flux[filter] = (wave_bin_width * (np.divide(np.multiply(area, thru),
+                             dd_wavedist) / constants.h.to(u.erg * u.s))).to(1 / u.s)
+
+        # Sum the flux in the filter
+        theo_flux_sum[filter] = np.sum(theo_flux[filter])
+
+        # Computed flux
+        gauss_flux[filter] = bandpass.compute_count_rate(thru, filter)
+
+        # Test that proper results (within 2%) are returned for select bands.
+        assert np.isclose(theo_flux_sum[filter].value, gauss_flux[filter], rtol=2.0e-02)
+
+    # Create log entry and artifacts
+    log.info('DMS233: integrated over an input spectra in physical units to derive the number of photons / s.')
+
+    artifactdir = os.environ.get('TEST_ARTIFACT_DIR', None)
+    if artifactdir is not None:
+        af = asdf.AsdfFile()
+        af.tree = {'theo_flux': theo_flux,
+                   'theo_flux_sum': theo_flux_sum,
+                   'gauss_flux': gauss_flux,
+                   'thru': thru}
+        af.write_to(os.path.join(artifactdir, 'dms233.asdf'))
+
+
+@pytest.mark.parametrize("filter", ['F062', 'F087', 'F106', 'F129', 'F158', 'F184', 'F213', 'F146'])
+def test_AB_convert_flux_to_counts(filter):
+    # AB Zero Test
+    abfv = 3631e-23 * u.erg / (u.s * u.cm ** 2 * u.hertz)
+
+    effarea = bandpass.read_gsfc_effarea()
+    wavedist = effarea['Wave'] * u.micron
+
+    flux = abfv * np.ones(len(wavedist))
+    gauss_flux = bandpass.compute_count_rate(flux, filter)
```

Comment:
Why gauss_flux?

Reply:
Holdover from copying a line from a different test - I have renamed it.